src/meta/test/BalanceIntegrationTest.cpp

```diff
@@ -22,6 +26,114 @@ TEST(BalanceIntegrationTest, SimpleTest) {
     LOG(INFO) << "Start storage server on " << sc->port_;
 }
 
+TEST(BalanceIntegrationTest, LeaderBalanceTest) {
+    FLAGS_load_data_interval_secs = 1;
+    FLAGS_heartbeat_interval_secs = 1;
+    FLAGS_heartbeat_interval = 1;
+    fs::TempDir rootPath("/tmp/balance_integration_test.XXXXXX");
+    IPv4 localIp;
+    network::NetworkUtils::ipv4ToInt("127.0.0.1", localIp);
+
+    uint32_t localMetaPort = network::NetworkUtils::getAvailablePort();
+    LOG(INFO) << "Start meta server....";
+    std::string metaPath = folly::stringPrintf("%s/meta", rootPath.path());
+    auto metaServerContext = meta::TestUtils::mockMetaServer(localMetaPort, metaPath.c_str());
+    localMetaPort = metaServerContext->port_;
+
+    auto adminClient = std::make_unique<AdminClient>(metaServerContext->kvStore_.get());
+    Balancer balancer(metaServerContext->kvStore_.get(), std::move(adminClient));
+
+    LOG(INFO) << "Create meta client...";
+    auto threadPool = std::make_shared<folly::IOThreadPoolExecutor>(10);
+    std::vector<HostAddr> metaAddr = {HostAddr(localIp, localMetaPort)};
+
+    uint32_t tempDataPort = network::NetworkUtils::getAvailablePort();
+    HostAddr tempDataAddr(localIp, tempDataPort);
+    auto clusterMan = std::make_unique<nebula::meta::ClusterManager>("", "");
+    auto mClient = std::make_unique<meta::MetaClient>(threadPool, metaAddr, tempDataAddr,
+                                                      clusterMan.get(), false);
+
+    auto ret = mClient->addHosts({tempDataAddr}).get();
+    ASSERT_TRUE(ret.ok());
+    mClient->waitForMetadReady();
+    ret = mClient->removeHosts({tempDataAddr}).get();
+    ASSERT_TRUE(ret.ok());
+
+    int partition = 9;
+    int replica = 3;
```

Comment:
let's test 5 instead of 3 to make the case more complicated.
partition%relica!=0 is better.

Reply:
It would be test in BalancerTest